alerts/cluster-monitoring-operator/PrometheusScrapeBodySizeLimitHit.md

```diff
@@ -0,0 +1,120 @@
+# PrometheusScrapeBodySizeLimitHit
+
+## Meaning
+
+The alert `PrometheusScrapeBodySizeLimitHit` is triggered when at least one of
+Prometheus' scrape targets replies with a response body larger than the
+configured `body_size_limit`.
+By default there is no limit on body size of the scrape targets.
+When set, this limit prevents Prometheus from consuming excessive amounts of memory
+when scrape targets return a response that is deemed too large.
+This must be set explicitly by the cluster administrator.
+
+## Impact
+
+Metrics coming from targets responding with a body exceeding the configured
+size limit aren't ingested by Prometheus. The targets will be considered as being
+down and they will have their `up` metric set to 0 which may also trigger the
+`TargetDown` alert.
+
+## Diagnosis
+
+First, we need to check the value of the configured body size limit.
+
+This value is located in the configmap `cluster-monitoring-config` in the
+namespace `openshift-monitoring`.
+
+To check the value in configmap, use this command:
+```bash
+oc get cm -n openshift-monitoring cluster-monitoring-config -o yaml | grep enforcedBodySizeLimit
+```
+
+If the value is too small, we need to set it to a larger number. Please refer to
+[Mitigation](#mitigation) on how to change the value.
+
+If the value is correct, we need to find out which target is responding with a body
+too large for scraping.
+
+In the Openshift Web UI, go to the Observe > Targets page and check which
+targets are down.
+
+If we need more information than the UI provides, we can query the Prometheus API
+endpoint `/api/v1/targets`.
+In the field `data.activeTargets` search for those that have `health` field
+different than `up` and check for the `lastError` field for confirmation:
+
+1. Get the name of the secret containing the token of service account `prometheus-k8s`.
+We use the following command and check for the name  `prometheus-k8s-token-[a-z]+`
+under section `secret`. In the example below, the secret name is `prometheus-k8s-token-hxg8v`.
```

Comment:
due to change, see from https://bugzilla.redhat.com/show_bug.cgi?id=2093780
there is not prometheus-k8s-token secret in sa prometheus-k8s
we could use
`token=$(oc create token prometheus-k8s -n openshift-monitoring)`

Reply:
Thanks :) 
`oc get sa` does not show token secret any more. The name of token secret is still shown by `oc describe sa`. The PR is updated accordingly.